k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -140,28 +141,32 @@
     private final ConcurrentHashMap<Account, Pusher> pushers = new ConcurrentHashMap<>();
     private final ExecutorService threadPool = Executors.newCachedThreadPool();
     private final MemorizingMessagingListener memorizingMessagingListener = new MemorizingMessagingListener();
+    private final TransportProvider transportProvider;
 
 
     private MessagingListener checkMailListener = null;
     private volatile boolean stopped = false;
 
 
-    public synchronized static MessagingController getInstance(Context context) {
+    public static synchronized MessagingController getInstance(Context context) {
         if (inst == null) {
             Context appContext = context.getApplicationContext();
             NotificationController notificationController = NotificationController.newInstance(appContext);
             Contacts contacts = Contacts.getInstance(context);
-            inst = new MessagingController(appContext, notificationController, contacts);
+            TransportProvider transportProvider = TransportProvider.getInstance();
+            inst = new MessagingController(appContext, notificationController, contacts, transportProvider);
         }
         return inst;
     }
 
 
     @VisibleForTesting
-    MessagingController(Context context, NotificationController notificationController, Contacts contacts) {
+    MessagingController(Context context, NotificationController notificationController,
+                        Contacts contacts, TransportProvider transportProvider) {
```

Comment:
Please import the [K-9 code style settings](https://github.com/k9mail/k-9/tree/master/tools/android-studio) to avoid using the wrong indentation rules.